### ISSUE TITLE
Stuck at "Calculating earning for each miner"

### ISSUE THREAD

**Issue by Shahnewaz1996** [MultiPoolMiner/MultiPoolMiner#2442]
Hi there returning MPM user here.
I could not get the miner running after completing my setup, and I had to do the fix mentioned here:
https://github.com/MultiPoolMiner/MultiPoolMiner/issues/2382#issuecomment-549949736
And here to solve the "No pools available" issue: https://github.com/MultiPoolMiner/MultiPoolMiner/issues/2405#issuecomment-533481080
Now I've run into another issue. I cannot get the miner to do anything after initialization. It won't go past this:

2019-11-10 19:54:57 INFO: Starting MultiPoolMiner® v3.5.2 © 2017-2019 MultiPoolMiner.io
2019-11-10 19:55:00 INFO: Web dashboard and API (version 0.96) running on http://localhost:3999.
2019-11-10 19:55:00 INFO: Mining for you. Donation run will start in 00 hour(s) 50 minute(s).
2019-11-10 19:55:00 INFO: Miner status reporting process started.
2019-11-10 19:55:01 INFO: Loading pool information (MiningPoolHub-Algo) - this may take a minute or two.
2019-11-10 19:55:02 INFO: Waiting for pool information.
2019-11-10 19:55:03 INFO: Selecting best pool for each algorithm.
2019-11-10 19:55:04 INFO: Loading saved statistics.
2019-11-10 19:55:04 INFO: Getting miner information.
2019-11-10 19:55:07 INFO: Updating exchange rates from CryptoCompare.
2019-11-10 19:55:08 INFO: Calculating earning for each miner.

There is no indication that nothing has crashed, no extra/missing window, and I cannot tell it is doing anything in the background.

Software info:
Windows 10 Pro v1903
PowerShell v6.2.3
Radeon Software v19.9.2

**Comment by UselessGuru** [comment]
When this happens please press <Ctrl><Break> in the main CMD window.

At what line will the debugger break into the code?
Please post the screen output for further analysys, Thank you!

**Comment by Shahnewaz1996** [comment]
The main window doesn't go beyond: "Transcript started, output file is .\Logs\MultiPoolMiner_2019-11-11_08-32-45.txt"

**Comment by UselessGuru** [comment]
@Shahnewaz996 

There was a typo in my previous post.
Correct. When this happens please press \<Ctrl\>\<Break\> in the main CMD window.

**Comment by Shahnewaz1996** [comment]
Thanks for the reply. Here is the message I receive after entering debug mode. Same message, same position regardless of the comment on MultiPoolMiner.ps1

```
Entering debug mode. Use h or ? for help.

At C:\MultiPoolMiner\Reader.ps1:16 char:5
+     Get-Job | Where-Object {$_ | Receive-Job -Keep; $Active += @{Id = ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Using Display call stack command:
```
Command       Arguments                                Location
-------       ---------                                --------
Reader.ps1                                             Reader.ps1: line 16
<ScriptBlock> {{=$null}, {=$null}, {=$null}, {=$null}} <No file>
```

Using the list command:
```
   11:          Where-Object SideIndicator -EQ "=>" |
   12:          ForEach-Object {$Active += @{Id = (Start-Job ([ScriptBlock]::Create("Get-Content '$(Convert-Path ".\Logs\$($_.InputObject)")' -Wait$(if($QuickStart){" -Tail 1000"}) -Encoding UTF8")) -Name $_.InputObject).Id; Time = (Get-Date).ToUniversalTime()}}
   13:
   14:      Start-Sleep 1
   15:
   16:*     Get-Job | Where-Object {$_ | Receive-Job -Keep; $Active += @{Id = $_.Id; Time = (Get-Date).ToUniversalTime()}} | Select-Object -First 1 | Receive-Job | Where-Object {$_}
   17:
   18:      $Active = @($Active | Where-Object Time -ge (Get-Date).ToUniversalTime().AddMinutes(-10))
   19:      Get-Job | Where-Object {-not ($Active | Where-Object Id -EQ $_.Id)} | Remove-Job -Force
   20:  }
   21:
```

**Comment by UselessGuru** [comment]
@Shahnewaz1996 

I think you pressed \<Ctrl\>\<Break\> in the wrong window (the log reader).
You should press it in the main MPM cmd window.

Alternatively you can use Snaketail as log reader (I had it included as an option, but @Aaronsace removed it). [Download snaketail.exe from here ](https://github.com/snakefoot/snaketail-net/releases)(version 1.9.6 is fine) and copy the exe to the MPM directory.
This will most likely not fix your issue, but you'll find it easier to debug the correct window ;-)

**Comment by Shahnewaz1996** [comment]
The only window where Ctrl + Break works is on the window titled "C:\Program Files\PowerShell\6\pwsh.exe". The other window is simply stuck at "Transcript started, output file is..."

**Comment by UselessGuru** [comment]
There is another CMD window - probably minimzed in the task bar.

**Comment by Shahnewaz1996** [comment]
This is what's on my current desktop.
![Snipaste_2019-11-13_12-10-11](https://user-images.githubusercontent.com/34409006/68786913-93f77b80-060e-11ea-9bd0-bbb2cf7bb519.png)

**Comment by UselessGuru** [comment]
Hm...

Please try the snaketail version. It does not break anything. Just copy the exe and reatart MPM.

**Comment by Shahnewaz1996** [comment]
Now the Powershell window no longer appears. Ctrl + Break still does not work on the CMD window.
![Snipaste_2019-11-13_12-35-50](https://user-images.githubusercontent.com/34409006/68788815-2b120280-0612-11ea-82e8-9a1baeb7f0ef.png)

**Comment by UselessGuru** [comment]
??? Of course the XML should go in the main MPM folder - i thought the SnakeTail addon was still in the zip file. I'll make that available somehow. 

I am very puzzled about your issue. You should be able to break into the CMD window. 
Have you tried commenting out the lines mentioned here:
https://github.com/MultiPoolMiner/MultiPoolMiner/issues/2382#issuecomment-549949736
???

**Comment by Shahnewaz1996** [comment]
The program does gets stuck as https://github.com/MultiPoolMiner/MultiPoolMiner/issues/2382#issuecomment-524924657 with the lines un-commented. Ctrl + Break does not work on any window. Running as admin here does not help.

Only commenting those lines out lets the application to proceed, but as far as shown here. Admin privileges do not help. Ctrl + Break does not work on the main window, only on PowerShell.

SnakeTail works now, but I'm not sure what to do with it.

**Comment by UselessGuru** [comment]
@Shahnewaz1996 

The only way I see to troubleshoot further is to run MPM in a debugging environment.

1. Download 'Visual Studio Code' https://code.visualstudio.com/download
2. In VSC load the MultipoolMiner.ps1 file in the MPM folder
3. On the first line `'using module .\Include.psm1'` press F9 to define a breakpoint
4. Run MultiPoolMiner.ps1 with \<F5\>
5. Step through the code line by line. Go to the next line by pressing \<F10\>

**Comment by UselessGuru** [comment]
@Shahnewaz1996 

Any update?

**Comment by Shahnewaz1996** [comment]
Without any changes I get stuck at line 189 in the "Unblock files" section.
Commenting them out, I can get all the way through till it starts complaining about "No pools available." (line 549).

**Comment by UselessGuru** [comment]
@Shahnewaz1996 

What is your powershell execution policy?? This can be checked with `Get-ExecutionPolicy` in a powershell command window.

If is  not 'Remotesigned' then change config with 
`Set-ExecutionPolicy Remotesigned` (this requires administrative privileges).

As for the 'No Pools' message: 
There are two instances in the code, approx. line 447 & line 594. Which one is printing it?

I assume a problem specific to your config file / start cmd file. Could you please post your start files & 'config.txt'?

Thank  you!

**Comment by BoDaly** [comment]
@UselessGuru I am having this same issue,
I have tried everything including taking a known working instance of MPM and got the same issue.
I can get the debug to work as long as i push it before the
`Transcript started, output file is .\Logs\MultiPoolMiner_2020-01-31_17-35-39.txt`

I can also see a single debug after the `Transcript started, output file is .\Logs\MultiPoolMiner_2020-01-31_17-35-39.txt` if i hit the ctrl+break and then hit ctrl+c which kills the run. it gives me this:
`At line:130 char:13
+             throw
+             ~~~~~
[DBG]: PS C:\Users\Great Fox\Desktop\MultiPoolMiner-3.5.5>>     `

If i step through in debug I get to a point of:
![image](https://user-images.githubusercontent.com/31381847/73585019-bb155a80-4451-11ea-8c82-e0937aa17d2a.png)

anything else you need? I have also gone through all of the stuff you had mentioned previously on this thread.

**Comment by BoDaly** [comment]
and just for good measure this is right before that in the step process.
![image](https://user-images.githubusercontent.com/31381847/73585089-966db280-4452-11ea-8ae1-f5e366831eab.png)

**Comment by compil3** [comment]
@BoDaly did you ever resolve the issue?

**Comment by Shahnewaz1996** [comment]
Somewhere down the line this issue is now (presumably) fixed for me.
I don't have the old setup anymore, so I can't double-check and see what the issue was with the old configuration.
The program works properly now.
